src/phases/turn-start-phase.ts

```diff
@@ -17,18 +17,59 @@ import { PostTurnStatusEffectPhase } from "./post-turn-status-effect-phase";
 import { SwitchSummonPhase } from "./switch-summon-phase";
 import { TurnEndPhase } from "./turn-end-phase";
 import { WeatherEffectPhase } from "./weather-effect-phase";
+import { BattlerIndex } from "#app/battle";
+import { TrickRoomTag } from "#app/data/arena-tag";
 
 export class TurnStartPhase extends FieldPhase {
   constructor(scene: BattleScene) {
     super(scene);
   }
 
-  start() {
-    super.start();
+  /**
+   * This orders the active Pokemon on the field by speed into an BattlerIndex array and returns that array.
+   * It also checks for Trick Room and reverses the array if it is present.
+   * @return {@linkcode BattlerIndex[]}
+   */
+  getSpeedOrder(): BattlerIndex[] {
+    const playerField = this.scene.getPlayerField().filter(p => p.isActive()) as Pokemon[];
+    const enemyField = this.scene.getEnemyField().filter(p => p.isActive()) as Pokemon[];
+
+    // We shuffle the list before sorting so speed ties produce random results
+    let orderedTargets: Pokemon[] = playerField.concat(enemyField);
+    // We seed it with the current turn to prevent an inconsistency where it
+    // was varying based on how long since you last reloaded
+    this.scene.executeWithSeedOffset(() => {
+      orderedTargets = Utils.randSeedShuffle(orderedTargets);
+    }, this.scene.currentBattle.turn, this.scene.waveSeed);
+
+    orderedTargets.sort((a: Pokemon, b: Pokemon) => {
+      const aSpeed = a?.getBattleStat(Stat.SPD) || 0;
+      const bSpeed = b?.getBattleStat(Stat.SPD) || 0;
+
+      return bSpeed - aSpeed;
+    });
 
-    const field = this.scene.getField();
-    const order = this.getOrder();
+    // Next, a check for Trick Room is applied. If Trick Room is present, the order is reversed.
+    const speedReversed = new Utils.BooleanHolder(false);
+    this.scene.arena.applyTags(TrickRoomTag, speedReversed);
 
+    if (speedReversed.value) {
+      orderedTargets = orderedTargets.reverse();
+    }
+
+    return orderedTargets.map(t => t.getFieldIndex() + (!t.isPlayer() ? BattlerIndex.ENEMY : BattlerIndex.PLAYER));
+  }
+
+  /**
+   * This takes the result of getSpeedOrder and applies priority / bypass speed attributes to it.
+   * This also considers the priority levels of various commands and changes the result of getSpeedOrder based on such.
+   * @return {@linkcode BattlerIndex[]} this determines the timeline of the current turn
```

Comment:
```suggestion
   * @returns {@linkcode BattlerIndex[]} the final sequence of commands for this turn
```
A description like this for `getSpeedOrder`'s `@returns` would also be nice.